c7n/query.py

```diff
@@ -806,8 +846,36 @@ class TypeInfo(metaclass=TypeMeta):
 
     **Arn handling / generation metadata - Optional**
 
-    :param arn: Arn resource attribute, when describe format has arn
-    :param arn_type: Type, used for arn construction, also required for universal tag augment
+    :param arn: Defines a field in the resource definition that contains the ARN value, when
+        the resource has an ARM..
+
+        This value is accessed used by the 'get_arns(..)' fn on the super-class
+        QueryResourceManager. This value must be a simple field name and cannot be a path.
+
+        If this value is not defined then 'get_arns' contains fallback logic.
+        - First fallback logic is to look at what's defined in the 'id' field of the resource.
+        If the value of the "id" field starts with "arn:" then that value is used as the arn.
+        - Otherwise, an attempt at generating (guessing!) the ARN by assembling it from
+        various fields and runtime values based on a recipe defined in 'generate_arn()' on
+        the super-class QueryResourceManager.
+
+        If you aren't going to define the "arn" field and can't rely on the "id" to be an
+        ARN then you might get lucky that "generate_arn" works for your resource type.
+        However, failing that then you should override "get_arns" function entirely and
+        implement your own logic.
+
+        Testing: Whatever approach you use (above) you REALLY SHOULD (!!!) include a unit
+        test that verifies that "get_arns" yields the right shape of ARNs for your resources.
+        This test should be implemented as an additional assertion within the unit tests
+        you'll be already planning to write.
```

Comment:
Bit of a judgement call here, but some of this feels like it goes beyond explaining the parameters and into "recommendations for creating new resources". This extra level of detail feels more appropriate in something like the [Adding a New Resource](https://cloudcustodian.io/docs/aws/contribute.html#adding-new-aws-resources) documentation that is focused on helping contributors.

Reply:
The Intention is to verify the API calls that my config and code is making cc perform, and. Not testing boto. Adding this allows.me to be sure that my apps behave as Intended in a way that placebo does not. For example placebo alone is incapable of doing verification of the values passed to the APIs or even the number of calls made really. This is simply a better mock of the boto API to confirm behaviour

Reply:
I understand your point about moving the comment to the "new resource" content  . I was just concerned about it getting separated from the related comments in the preceding paragraph.

Thoughts?